tests/Service/DatabaseManagerTest.php

```diff
@@ -140,6 +161,33 @@ public function testIsCreatingAndRollingBackTransaction(): void
         $this->assertTrue($db->rollBack());
     }
 
+    public function testIsCreatingAndRollingBackTransactionUsingExecuteTransaction(): void
+    {
+        $this->expectException(DatabaseException::class);
```

Comment:
You should put this right before the code that will throw the exception. Otherwise, anything that throws the exception before hand would pass the test.

Reply:
Ok.